### ISSUE TITLE
`mock` is failing to install extras in Python 3.6

### ISSUE THREAD

**Issue by rowillia** [pypa/setuptools#800]
Howdy,

I'm attempting to make a change to the `mock` package in Python, and it appears their CI is broken in Python 3.6 only.  The problem appears to originate from setuptools, so I was hoping you folks might have more insight.

Here's the last time the `mock` jobs passed:
https://travis-ci.org/testing-cabal/mock/jobs/153850304 
Here's where they started failing:
https://travis-ci.org/testing-cabal/mock/jobs/159121114

Here's the core of the issue.  AFAICT in Python 3.6 setuptools isn't parsing (or is incorrectly parsing) `setup.cfg`.

```
1.43s$ pip install -U .[docs,test]
Processing /home/travis/build/testing-cabal/mock
  mock 2.0.1.dev2 does not provide the extra 'docs'
  mock 2.0.1.dev2 does not provide the extra 'test'
Installing collected packages: mock
  Found existing installation: mock 2.0.0
    Uninstalling mock-2.0.0:
      Successfully uninstalled mock-2.0.0
  Running setup.py install for mock ... done
Successfully installed mock-2.0.1.dev2
```

I attempted to force travis to pin to the latest version of setuptools and had the same issue (only in Python 3.6, not 3.5 - https://travis-ci.org/testing-cabal/mock/builds/162838672)

Interestingly, when this build went from "passing" to "failing", both were running `Python 3.6.0a4+`, so I don't think it's an issue with Python 3.6 itself, but if you think it is I'll file an issue at bugs.python.org.

**Comment by rowillia** [comment]
Filed http://bugs.python.org/issue28279.  Since 3.6 is in beta it's better to raise these issues with the core devs sooner rather than later 😄

**Comment by jaraco** [comment]
I'm unable to replicate this with the stable release of mock:

```
$ python -m pip install -t foo --no-binary :all: 'mock[docs,test]'
Collecting mock[docs,test]
  Downloading mock-2.0.0.tar.gz (73kB)
    100% |████████████████████████████████| 81kB 567kB/s 
Collecting pbr>=0.11 (from mock[docs,test])
  Downloading pbr-1.10.0.tar.gz (111kB)
    100% |████████████████████████████████| 112kB 244kB/s 
Collecting six>=1.9 (from mock[docs,test])
  Downloading six-1.10.0.tar.gz
Collecting sphinx (from mock[docs,test])
  Downloading Sphinx-1.4.8.tar.gz (4.5MB)
    100% |████████████████████████████████| 4.5MB 220kB/s 
Collecting unittest2>=1.1.0 (from mock[docs,test])
  Downloading unittest2-1.1.0.tar.gz (81kB)
    100% |████████████████████████████████| 81kB 701kB/s 
Collecting Jinja2>=2.3 (from sphinx->mock[docs,test])
  Downloading Jinja2-2.8.tar.gz (357kB)
    100% |████████████████████████████████| 358kB 354kB/s 
Collecting Pygments>=2.0 (from sphinx->mock[docs,test])
  Downloading Pygments-2.1.3.tar.gz (2.1MB)
    100% |████████████████████████████████| 2.1MB 465kB/s 
Collecting docutils>=0.11 (from sphinx->mock[docs,test])
  Using cached docutils-0.12.tar.gz
Collecting snowballstemmer>=1.1 (from sphinx->mock[docs,test])
  Downloading snowballstemmer-1.2.1.tar.gz (49kB)
    100% |████████████████████████████████| 51kB 1.2MB/s 
Collecting babel!=2.0,>=1.3 (from sphinx->mock[docs,test])
  Downloading Babel-2.3.4.tar.gz (6.9MB)
    100% |████████████████████████████████| 6.9MB 158kB/s 
Collecting alabaster<0.8,>=0.7 (from sphinx->mock[docs,test])
  Downloading alabaster-0.7.9.tar.gz
Collecting imagesize (from sphinx->mock[docs,test])
  Downloading imagesize-0.7.1.tar.gz
Collecting argparse (from unittest2>=1.1.0->mock[docs,test])
  Downloading argparse-1.4.0.tar.gz (70kB)
    100% |████████████████████████████████| 71kB 612kB/s 
Collecting traceback2 (from unittest2>=1.1.0->mock[docs,test])
  Downloading traceback2-1.4.0.tar.gz
Collecting MarkupSafe (from Jinja2>=2.3->sphinx->mock[docs,test])
  Using cached MarkupSafe-0.23.tar.gz
Collecting pytz>=0a (from babel!=2.0,>=1.3->sphinx->mock[docs,test])
  Downloading pytz-2016.7.zip (497kB)
    100% |████████████████████████████████| 501kB 314kB/s 
Collecting linecache2 (from traceback2->unittest2>=1.1.0->mock[docs,test])
  Downloading linecache2-1.0.0.tar.gz
Installing collected packages: pbr, six, MarkupSafe, Jinja2, Pygments, docutils, snowballstemmer, pytz, babel, alabaster, imagesize, sphinx, argparse, linecache2, traceback2, unittest2, mock
  Running setup.py install for pbr ... done
  Running setup.py install for six ... done
  Running setup.py install for MarkupSafe ... done
  Running setup.py install for Jinja2 ... done
  Running setup.py install for Pygments ... done
  Running setup.py install for docutils ... done
  Running setup.py install for snowballstemmer ... done
  Running setup.py install for pytz ... done
  Running setup.py install for babel ... done
  Running setup.py install for alabaster ... done
  Running setup.py install for imagesize ... done
  Running setup.py install for sphinx ... done
  Running setup.py install for argparse ... done
  Running setup.py install for linecache2 ... done
  Running setup.py install for traceback2 ... done
  Running setup.py install for unittest2 ... done
  Running setup.py install for mock ... done
Successfully installed Jinja2 MarkupSafe Pygments alabaster argparse babel docutils imagesize linecache2 mock pbr pytz six-1.10.0 snowballstemmer sphinx traceback2 unittest2
$ python -V
Python 3.6.0b2
```

So either the issue has been fixed in Python 3.6.0b2 or the issue exists in the unreleased mock codebase or the issue is unique to the Travis environment. In any case, I'm not aware of any changes in Setuptools nor Python/distutils that might implicate this failure, and there's not enough context for either project to address the issue. I suggest if the issue persists that you dig deeper into the mock project and identify where the behavior diverges from your expectation. That is, if you can find that setuptools isn't generating a setup.cfg properly or Python isn't parsing it properly, then we can address it in upstream tickets.
